scripts/singularity_cmd_via_docker

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+#emacs: -*- mode: shell-script; c-basic-offset: 4; tab-width: 4; indent-tabs-mode: t -*- 
+#ex: set sts=4 ts=4 sw=4 noet:
+#
+# A helper to run/exec singularity images using Docker with sanitization:
+#  - fake HOME with minimalistic .bashrc and .gitconfig
+#  - dedicated /tmp /var/tmp
+#  - cleansed environment variables
+# while bind mounting (and starting from) current directory.
+#
+# COPYRIGHT: Yaroslav Halchenko 2019
+#
+# LICENSE: MIT
+#
+#  Permission is hereby granted, free of charge, to any person obtaining a copy
+#  of this software and associated documentation files (the "Software"), to deal
+#  in the Software without restriction, including without limitation the rights
+#  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+#  copies of the Software, and to permit persons to whom the Software is
+#  furnished to do so, subject to the following conditions:
+#
+#  The above copyright notice and this permission notice shall be included in
+#  all copies or substantial portions of the Software.
+#
+#  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+#  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+#  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+#  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+#  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+#  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+#  THE SOFTWARE.
+#
+
+set -eu
+
+function info() {
+	: # echo -e "I: $@" >&2
+}
+
+thisdir=$(dirname $0| xargs readlink -f)
+updir=$(dirname "$thisdir")
+
+cmd="${SINGULARITY_CMD:-$1}"; shift
+
+if [ ! -z "${DATALAD_CONTAINER_NAME:-}" ]; then
+	export SINGULARITYENV_DATALAD_CONTAINER_NAME="$DATALAD_CONTAINER_NAME"
+fi
+
+USER_ID=$(id -u)
+GROUP_ID=$(id -g)
+
+# singularity bind mounts system /tmp, which might result in side-effects
+# Create a dedicated temporary directory to be removed upon completion
+tmpdir=$(mktemp -d --suffix=singtmp)
+info "created temp dir $tmpdir"
+trap "rm -fr '$tmpdir' && info 'removed temp dir $tmpdir'" exit
+
+docker run -ti --privileged --rm \
+      -v $PWD:/working-dir \
+	  -v $tmpdir:/working-temp \
+      -w /working-dir \
+      singularityware/singularity:latest \
+      "$cmd" -e -c -W /working-temp -H "$updir/binds/HOME" \
```

Comment:
I think that the [singularity_cmd](https://github.com/ReproNim/containers/blob/master/scripts/singularity_cmd#L66) should get those `$BHOME` and `$tmpdir` exposed via some environment variable(s) (e.g. `REPRONIM_CONTAINERS_BINDS`) so this script would just bind mount them...
Overall think about "interaction" between that `singularity_cmd` and this script, or just moving this "docker run ... singularity" invocation instead of plain "singularity" call within that script,  to be triggered via `SINGULARITY_VIA_DOCKER` env variable, which is if empty decided based on OS (non-Linux - use docker).